lib/presentation/widget/upload_request_inside/upload_request_inside_navigator_widget.dart

```diff
@@ -0,0 +1,137 @@
+// the words “You are using the Free and Open Source version of LinShare™, powered by
+// Linagora © 2009–2020. Contribute to Linshare R&D by subscribing to an Enterprise
+// offer!”. You must also retain the latter notice in all asynchronous messages such as
+// e-mails sent with the Program, (ii) retain all hypertext links between LinShare and
+// http://www.linshare.org, between linagora.com and Linagora, and (iii) refrain from
+// infringing Linagora intellectual property rights over its trademarks and commercial
+// brands. Other Additional Terms apply, see
+// <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf>
+// for more details.
+// This program is distributed in the hope that it will be useful, but WITHOUT ANY
+// WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+// FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for
+// more details.
+// You should have received a copy of the GNU Affero General Public License and its
+// applicable Additional Terms for LinShare along with this program. If not, see
+// <http://www.gnu.org/licenses/> for the GNU Affero General Public License version
+//  3 and <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf> for
+//  the Additional Terms applicable to LinShare software.
+
+
+import 'package:domain/domain.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:linshare_flutter_app/presentation/widget/upload_request_inside/upload_request_document_arguments.dart';
+import 'package:linshare_flutter_app/presentation/widget/upload_request_inside/upload_request_document_type.dart';
+import 'package:linshare_flutter_app/presentation/widget/upload_request_inside/upload_request_inside_widget.dart';
+import 'package:rxdart/rxdart.dart';
+
+import '../../util/data_structure/stack.dart' as structure;
+
+typedef OnBackUploadRequestClickedCallback = void Function();
+typedef OnNodeClickedCallback = void Function(UploadRequestGroup uploadRequestGroup);
+
+class UploadRequestNavigatorWidget extends StatefulWidget {
+
+  final UploadRequestGroup uploadRequestGroup;
+  final OnBackUploadRequestClickedCallback? onBackUploadRequestClickedCallback;
+
+  final BehaviorSubject<UploadRequestArguments>? currentNodeObservable;
+
+  UploadRequestNavigatorWidget(
+    Key key,
+    this.uploadRequestGroup,
+    {
+      this.onBackUploadRequestClickedCallback,
+      this.currentNodeObservable
+    }
+  ) : super(key: key);
+
+  @override
+  UploadRequestNavigatorWidgetState createState() => UploadRequestNavigatorWidgetState();
+}
+
+class UploadRequestNavigatorWidgetState extends State<UploadRequestNavigatorWidget> {
+
+  final GlobalKey<NavigatorState> navigatorKey = GlobalKey();
+
+  final structure.Stack<UploadRequestArguments> argumentStack = structure.Stack();
+  UploadRequestArguments? get currentPageData => argumentStack.peek();
```

Comment:
Do you have any documents I can refer to for this kind of stack ? It's so interesting

Reply:
It's only the normal stack. There is a library [here](https://pub.dev/packages/stack), but it's not necessary to use a library. We defined here: https://github.com/linagora/linshare-mobile-flutter-app/blob/master/lib/presentation/util/data_structure/stack.dart